rust/parquet/src/arrow/array_reader.rs

```diff
@@ -196,11 +196,13 @@ impl<T: DataType> ArrayReader for PrimitiveArrayReader<T> {
                         .convert(self.record_reader.cast::<Int64Type>()),
                     _ => Err(general_err!("No conversion from parquet type to arrow type for timestamp with unit {:?}", unit)),
                 },
-                (ArrowType::Date32(_), PhysicalType::INT32) => {
-                    UInt32Converter::new().convert(self.record_reader.cast::<Int32Type>())
+                (ArrowType::Date32(unit), PhysicalType::INT32) => match unit {
+                    DateUnit::Day => Date32Converter::new().convert(self.record_reader.cast::<Int32Type>()),
```

Comment:
nit: long line. Please limit width to 90 chars. Same below.

Reply:
Fixed.